src/test/objectstore/store_test.cc

```diff
@@ -7238,6 +7238,133 @@ TEST_P(StoreTestSpecificAUSize, BlobReuseOnOverwrite) {
   }
 }
 
+TEST_P(StoreTestSpecificAUSize, ZeroBlockDetectionSmall) {
+  if (string(GetParam()) != "bluestore")
+    return;
+  if (smr) {
```

Comment:
IIUC this patch is irrelevant to smr drive writing as long as test case performs append only. So may be it makes sense to apply this testing to smr drives - at least partially.